server/src/configs/compression_algorithm.rs

```diff
@@ -0,0 +1,104 @@
+use serde::{
+    de::{self, Deserializer, Visitor},
+    Deserialize, Serialize, Serializer,
+};
+use std::str::FromStr;
+
+// for now only those, in the future will add snappy, lz4, zstd (same as in confluent kafka) in addition to that
+// we should consider brotli as well.
+#[derive(Debug, PartialEq, Clone)]
+pub enum CompressionAlgorithm {
+    Producer,
+    Gzip,
+}
+
+impl FromStr for CompressionAlgorithm {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Producer" | "producer" => Ok(CompressionAlgorithm::Producer),
+            "Gzip" | "gzip" => Ok(CompressionAlgorithm::Gzip),
+            _ => Err(format!("Unknown compression type: {}", s)),
+        }
+    }
+}
+
+impl Serialize for CompressionAlgorithm {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match self {
+            CompressionAlgorithm::Producer => serializer.serialize_str("producer"),
+            CompressionAlgorithm::Gzip => serializer.serialize_str("gzip"),
+        }
+    }
+}
+struct CompressionAlgorithmVisitor;
+
+impl<'de> Visitor<'de> for CompressionAlgorithmVisitor {
+    type Value = CompressionAlgorithm;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_str("a valid compression type, check documentation for more information.")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: de::Error,
+    {
+        CompressionAlgorithm::from_str(value).map_err(de::Error::custom)
+    }
+}
+
+impl<'de> Deserialize<'de> for CompressionAlgorithm {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(CompressionAlgorithmVisitor)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+
+    #[test]
+    fn test_serialize() {
+        let producer_alg = CompressionAlgorithm::Producer;
+        let producer_serialized = serde_json::to_string(&producer_alg).unwrap();
```

Comment:
Should we use `serde_json`, as it seems that it's a basic mapping between an enum and string types? Maybe just the usage of `Into` and `From` traits would suffice? Unless it's supposed to be extended in the future and contain additional fields (e.g. as a struct).